tests/test_edp.py

```diff
@@ -0,0 +1,398 @@
+def record_rich(scrape_pos_page_body_rich):
+    """Return results generator from the EDP spider."""
+    spider = edp_spider.EDPSpider()
+    fake_resp = fake_response_from_file('edp/rich.xml')
+    fake_resp.meta["rich"] = True
+    node = get_node(spider, "//EDPSArticle", fake_resp)[0]
+    request = spider.parse_node(fake_resp, node).next()
+    response = HtmlResponse(
+        url=request.url,
+        request=request,
+        body=scrape_pos_page_body_rich,
+        **{'encoding': 'utf-8'}
+    )
+
+    return request.callback(response)
+
+
+def test_title_rich(record_rich):
+    """Test extracting title."""
+    title = "A representative sample of Be stars"
+    subtitle = "II. $K$ band spectroscopy"
+
+    assert 'title' in record_rich
+    assert 'subtitle' in record_rich
+    assert record_rich['title'] == title
+    assert record_rich['subtitle'] == subtitle
+
+
+def test_date_published_rich(record_rich):
+    """Test extracting date_published."""
+    date_published = "2000-01"
+
+    assert 'date_published' in record_rich
+    assert record_rich['date_published'] == date_published
+
+
+def test_collections_rich(record_rich):
+    """Test extracting collections."""
+    collections = ['HEP', 'Published']
+
+    assert 'collections' in record_rich
+    for coll in collections:
+        assert {"primary": coll} in record_rich['collections']
+
+
+def test_language_rich(record_rich):
+    """Test extracting language."""
+    assert 'language' not in record_rich
+
+
+def test_pages_rich(record_rich):
+    """Test page extraction."""
+    assert record_rich["page_nr"][0] == "13"
+    assert record_rich["journal_fpage"] == "65"
+    assert record_rich["journal_lpage"] == "77"
+
+
+def test_publication_info_rich(record_rich):
+    """Test extracting dois."""
+    journal_title = "Astronomy and Astrophysics Supplement Series"
+    journal_year = "2000"
+    journal_volume = "141"
+
+    assert 'journal_title' in record_rich
+    assert record_rich['journal_title'] == journal_title
+    assert 'journal_year' in record_rich
+    assert record_rich['journal_year'] == journal_year
+    assert 'journal_volume' in record_rich
+    assert record_rich['journal_volume'] == journal_volume
+
+
+def test_authors_rich(record_rich):
+    """Test authors."""
+    authors = ["Clark, J.S.", "Steele, I.A."]
+    surnames = ["Clark", "Steele"]
+    affiliations = [
+        "Astronomy Centre, CPES, University of Sussex, Brighton, BN1 9QH, UK",
+        "Astrophysics Research Institute, Liverpool John Moores University, Liverpool, L41 1LD, UK"
+    ]
+
+    assert 'authors' in record_rich
+    astr = record_rich['authors']
+    assert len(astr) == len(authors)
+    # here we are making sure order is kept
+    for index in range(len(authors)):
+        assert astr[index]['full_name'] == authors[index]
+        assert astr[index]['surname'] == surnames[index]
+        assert astr[index]["affiliations"][0]["value"] == affiliations[index]
+
+
+@pytest.fixture
```

Comment:
I'd recommend putting all the fixtures together, usually on top of the file, or in a separated module if they are reused on other tests (not sure if this is the case).